packages/amplify-cli/src/execution-manager.ts

```diff
@@ -39,6 +39,33 @@ export function isContainersEnabled(context) {
 }
 
 async function selectPluginForExecution(context: Context, pluginCandidates: PluginInfo[]): Promise<PluginInfo> {
+  const pluginCandidatesCategorySet = new Set<string>();
+  const overidedcategories = ['auth'];
+
+  pluginCandidates.forEach(plugin => {
+    pluginCandidatesCategorySet.add(plugin.manifest.name);
+  });
+
+  // overrided packageName format : @aws-amplify/amplify-category-<catgoryName>
+  const pluginName = pluginCandidatesCategorySet.values().next().value;
+  if (pluginCandidatesCategorySet.size == 1 && overidedcategories.includes(pluginName)) {
+    if (FeatureFlags.getBoolean(`overrides.${pluginName}`)) {
+      const overidedPlugin = pluginCandidates.find(plugin => {
+        return plugin.packageName === `@aws-amplify/amplify-category-${pluginName}`;
+      });
+      if (overidedPlugin !== undefined) {
+        return overidedPlugin;
+      }
+    } else {
+      const normalPlugin = pluginCandidates.find(plugin => {
```

Comment:
can we use better var naming here? pluginWithoutOverrides & pluginWithOverrides is one option.